packages/wallets/particle/src/adapter.ts

```diff
@@ -74,17 +108,29 @@ export class ParticleAdapter extends BaseMessageSignerWalletAdapter {
                 throw new WalletLoadError(error?.message, error);
             }
 
-            let wallet: SolanaWallet;
+            let particleNetwork: ParticleNetwork;
+
+            const authOptions: any = {};
+            if (this._config.preferredAuthType) {
+                authOptions.preferredAuthType = this._config.preferredAuthType;
+            }
+
             try {
-                wallet = new WalletClass(new ParticleClass(this._config?.config).auth);
+                particleNetwork = new ParticleClass(this._config?.config as Config);
+                if (!particleNetwork.auth.isLogin()) {
+                    await particleNetwork.auth.login(authOptions);
+                }
             } catch (error: any) {
                 throw new WalletConfigError(error?.message, error);
             }
 
+            this._particleNetwork = particleNetwork;
+
+            let wallet: SolanaWallet;
             try {
-                await wallet.connect();
+                wallet = new WalletClass(particleNetwork.auth);
             } catch (error: any) {
-                throw new WalletConnectionError(error?.message, error);
+                throw new WalletConfigError(error?.message, error);
```

Comment:
Can you describe a bit what this new code is doing? Is the `wallet` instance always connected? What about the case where `!particleNetwork.auth.isLogin()` is `false`?

Reply:
If `isLogin` returns false, then it initiates login through `particleNetwork.auth.login`, otherwise if it returns true, an assumption can be made that the user already has an open instance with Particle, in which this account is either used instead or the connection is already initiated. The changes here to how `wallet` is handled is a response to a shift in the architecture of `solana-wallet`